### ISSUE TITLE
Create Huginn iPhone / Android app

### ISSUE THREAD

**Issue by cantino** [huginn/huginn#166]
Interesting features:
- Send rough location updates throughout the day (via standard iPhone cell tower location updates) to Huginn
- Receive push notifications
- Send high-resolution location information for a time window after receiving a certain push notification, or at customizable times

Huginn agent ideas:
- Geofencing
- Send relevant traffic and public transit information depending on where you are
- Do intention detection based on common locations you visit and send relevant information
- Send emergency information, earthquake data, tsunami warnings, police info based on location
- Detect when you leave home or work and make an event

[![Bountysource](https://www.bountysource.com/badge/issue?issue_id=1413987)](https://www.bountysource.com/issues/1413987-create-huginn-iphone-android-app)

**Comment by stefuNz** [comment]
Android +1

**Comment by cantino** [comment]
I've added a bounty to this issue.  Perhaps React Native would be a good choice?

**Comment by jonathancross** [comment]
It would be good to define the absolute minimum requirements for this app, eg: most important use case. API seems like the first place to start. Pushover Agent support seems to have been added in https://github.com/cantino/huginn/pull/233

Any other updates relevant to API which could be used from an app?

**Comment by cantino** [comment]
Agreed that Huginn needs an API if someone is going to build mobile apps against it, so that would seem like the right place to start. Some Huginn endpoints can return JSON instead of HTML, so parts of the API would work now, but we don't have a versioned API or non-session auth yet. Something like Doorkeeper might make sense there.

**Comment by j-rewerts** [comment]
Has there been any further thoughts on this issue recently? I think location-based triggers would be a really powerful feature. 

@cantino How much work would it be to build a versioned API for Huginn in your estimation?

**Comment by cantino** [comment]
Not complicated but a lot of endpoints to build. I’m not very active with
Huginn anymore, so I wouldn’t be building it.

**Comment by j-rewerts** [comment]
This seems like a pretty popular project. Could we get some people to step up and start maintaining it?  If the previous active members are busy, maybe we could put a call out to new users?

**Comment by dsander** [comment]
@j-rewerts I think it would be possible to build an app without having a versioned API. We decided not to version Huginn itself years ago to keep the maintenance as low as possible. Instead of versioning we do our best to only introduce breaking changes to Agents and their configuration if there is no other way around. If we knew of applications that rely on the API we would apply the same logic to the API.

Because we don't have API keys the app would need to use the session cookie to authenticate the user. A bit annoying but doable, [here](https://huginnio.herokuapp.com/scenarios/19) is a sample Scenario which can be used to authenticate Huginn Agents against the current Huginn instance 😄 .

**Comment by vinaykapadia** [comment]
This would be awesome. Could also be used to send SMS (like the IFTTT app), so that I don't need to go through Twilio, and can just send it from my phone.

**Comment by AJ0070** [comment]
Is this issue still open/required?
